client/identity.go

```diff
@@ -21,26 +22,78 @@ type Identity struct {
 	LoomAddr       loom.Address
 }
 
-func CreateIdentity(name string, ethKey string, dappchainKey string, chainID string) (*Identity, error) {
-	mainnetPrivKey, err := crypto.HexToECDSA(strings.TrimPrefix(ethKey, "0x"))
-	if err != nil {
-		return nil, err
+func CreateRandomIdentity(chainID string) (*Identity, error) {
+    return CreateIdentity(nil, nil, chainID)
+}
+
+func CreateIdentity(hexKeyOrECDSA interface{}, signerOrKey interface{}, chainID string) (*Identity, error) {
```

Comment:
lets add some docs above this function, its not really clear its purpose

Reply:
This is interface abuse, this function should take `*ecdsa.PrivateKey` and `auth.Signer` as input, then if we don't have them already add utility functions to convert a string to `*ecdsa.PrivateKey` or `auth.Signer`.

Reply:
Right, I wanted to avoid adding utility functions on the end user, because if we want to create multiple identities such as in tests, this involves calling the utility functions multiple times externally, while instead this can be done internally with the interfaces.

Reply:
Can add higher level helper functions in tests.

Reply:
OK, np addressed then.